### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by hfiguiere**
Build with address sanitizer and run the tests

### CODE REVIEW

File: common/flatpak-transaction.c [hunk 1]
```diff
@@ -4075,7 +4075,7 @@ request_tokens_for_remote (FlatpakTransaction *self,
       g_autofree const char **refs_strv;
 
       g_variant_iter_init (&iter, tokens);
-      while (g_variant_iter_next (&iter, "{&s^a&s}", &token_for_refs, &refs_strv))
+      while (g_variant_iter_next (&iter, "{&s&a&s}", &token_for_refs, &refs_strv))
```
Comment (hfiguiere) [inline common/flatpak-transaction.c:4078]: this is actually invalid. Will fix.

**Comment by hfiguiere** [comment]
I guess that why we skipp the valgrind test?

CI is green now. There are some leaks left though, not sure how to address them.

**Comment by TingPing** [comment]
> I guess that why we skipp the valgrind test?

I think it was too slow, like beyond the limits of Github CI.

**Comment by hfiguiere** [comment]
yeah I saw after that we have asan without leaksanitizer.

**Comment by smcv** [comment]
> There are some leaks left though, not sure how to address them

#5690 fixed some more (in production code), and #5691 will fix more (in tests); reviews welcome. I think the only way to eliminate memory leaks is to work our way through like that, one test or a few tests at a time, until they all pass, and then add AddressSanitizer to the Github CI so that it won't regress.

**Comment by hfiguiere** [comment]
Sadly some of the remaining leaks in test haven't been fixed with your PR. 

I will file issue for them.

**Comment by smcv** [comment]
> Sadly some of the remaining leaks in test haven't been fixed with your PR

Yeah, it's a slow process - I only got as far as making `testlibrary` pass (with GLib 2.79.2, other versions might have different leaks).

**Pull request merged**